core/src/main/java/com/linecorp/armeria/server/RoutingPredicate.java

```diff
@@ -200,4 +293,47 @@ public String toString() {
                           .add("delegate", delegate)
                           .toString();
     }
+
+    private static final class NamedPredicate<T> {
+        private final String name;
+        private final Predicate<T> predicate;
```

Comment:
I don't see any big difference between `NamedPredicate` and `RoutingPredicate`.
Should we just use `RoutingPredicate` rather than introduce a similar class?